src/config.ts

```diff
@@ -43,3 +45,9 @@ export const defineConfig = (config: FullAuthConfig) => {
 	config.basePath = config.prefix
 	return config
 }
+
+export type UserAuthConfig =
+	| {
+			config: (ctx: APIContext | AstroGlobal | ActionAPIContext) => FullAuthConfig | Promise<FullAuthConfig>
+	  }
+	| FullAuthConfig
```

Comment:
What do you guys think about this?

```suggestion
export type LazyAuthConfig = (ctx: APIContext | AstroGlobal | ActionAPIContext) => FullAuthConfig | Promise<FullAuthConfig>

export type UserAuthConfig =
	| LazyAuthConfig
	| FullAuthConfig
```